packages/valory/skills/abstract_round_abci/abci_app_chain.py

```diff
@@ -39,6 +39,17 @@
 AbciAppTransitionMapping = Dict[AppState, AppState]
 
 
+def check_set_uniqueness(sets: Tuple) -> Optional[Any]:
+    """Checks that all elements in the set list are unique and not repeated among different sets"""
+    all_elements = set.union(*sets)
```

Comment:
why not just use `set.intersection`?

Reply:
I had left a comment to explain it but i don't see here anymore: set.intersection only works for chaining 2 apps. When chaining +2 apps, a round that is duplicated among 2 of those apps but not the other one won't appear on the intersection:
```
app1 = {a, b, c}
app2 = {c, d, e}
app3 = {f, g, h}
set.intersection(app1, app2, app3) = {}
```